torch/nn/modules/module.py

```diff
@@ -418,7 +418,15 @@ def forward(self, x):
     _backward_hooks: Dict[int, Callable]
     _is_full_backward_hook: Optional[bool]
     _forward_hooks: Dict[int, Callable]
+    # Marks whether the corresponding _forward_hooks accept kwargs or not. All
+    # As JIT does not support Set[int], this dict is used as a set, where all
+    # hooks represented in this dict accept kwargs.
+    _forward_hooks_with_kwargs: Dict[int, bool]
```

Comment:
One caveat is that when users directly access `_forward_hooks` to delete hooks, the counterpart in `_forward_hooks_with_kwargs` won't be deleted accordingly. This should be fine as `_forward_hooks_with_kwargs` only applies to new hooks so no BC issue?

@albanD @soulitzer @awgu @Padarn

Reply:
That sounds OK. But we may still want the corresponding entry in `_forward_hooks_with_kwargs` to be deleted if someone properly uses the handle to remove the hook.

Reply:
Yes the handle.remove() method should remove the entry in the _forward_hooks_with_kwargs dict.